lib/matplotlib/colors.py

```diff
@@ -3272,6 +3296,331 @@ def inverse(self, value):
+
+        Each component of the input is assigned to the constituent norm.
+
+        Parameters
+        ----------
+        value : array-like
+            Data to normalize, as tuple, scalar array or structured array.
+
+            - If tuple, must be of length `n_components`
+            - If scalar array, the first axis must be of length `n_components`
+            - If structured array, must have `n_components` fields.
+
+        clip : list of bools or bool or None, optional
+            Determines the behavior for mapping values outside the range
+            ``[vmin, vmax]``. See the description of the parameter *clip* in
+            `.Normalize`.
+            If ``None``, defaults to ``self.clip`` (which defaults to
+            ``False``).
+        structured_output : bool, optional
+
+            - If True, output is returned as a structured array
+            - If False, output is returned as a tuple of length `n_components`
+            - If None (default) output is returned in the same format as the input.
+
+        Returns
+        -------
+        tuple or `~numpy.ndarray`
+            Normalized input values`
+
+        Notes
+        -----
+        If not already initialized, ``self.vmin`` and ``self.vmax`` are
+        initialized using ``self.autoscale_None(value)``.
+        """
+        if clip is None:
+            clip = self.clip
+        elif not np.iterable(clip):
+            clip = [clip]*self.n_components
+
+        if structured_output is None:
+            if isinstance(value, np.ndarray) and value.dtype.fields is not None:
+                structured_output = True
+            else:
+                structured_output = False
+
+        value = self._iterable_components_in_data(value, self.n_components)
+
+        result = tuple(n(v, clip=c) for n, v, c in zip(self.norms, value, clip))
+
+        if structured_output:
+            result = self._ensure_multicomponent_data(result, self.n_components)
+
+        return result
+
+    def inverse(self, value):
+        """
+        Map the normalized value (i.e., index in the colormap) back to image data value.
+
+        Parameters
+        ----------
+        value
+            Normalized value, as tuple, scalar array or structured array.
+
+            - If tuple, must be of length `n_components`
+            - If scalar array, the first axis must be of length `n_components`
+            - If structured array, must have `n_components` fields.
+
+        """
+        value = self._iterable_components_in_data(value, self.n_components)
+        result = [n.inverse(v) for n, v in zip(self.norms, value)]
+        return result
+
+    def autoscale(self, A):
+        """
+        For each constituent norm, Set *vmin*, *vmax* to min, max of the corresponding
+        component in *A*.
+
+        Parameters
+        ----------
+        A
+            Data, must be of length `n_components` or be a structured array or scalar
+            with `n_components` fields.
```

Comment:
```suggestion
            Data, must be of length `n_components` or be a structured scalar or 
            structured array with `n_components` fields.
```
I would always repeat "structured" for this kind of expression because both scalar and array exist also in the unstructured form and only implying "structured" through the "and" can be misleading. Same for similar cases.